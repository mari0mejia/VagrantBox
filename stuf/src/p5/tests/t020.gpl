////
//Test four variants of if-statement, NO curly-braces
////
on akey {
  if(42 == 42)
    print("Test-A, they are equal.");
}

on akey {
  if(42 == 41)
    print("Test-B, should not see this message.");
}

on akey {
  if(42 == 41)
    print("Test-C, should not see this message.");
  else
    print("Test-C, they are not equal.");
}

on akey {
  if(42 == 42)
    print("Test-D, they are equal.");
  else
    print("Test-D, should not see this message.");
}

on akey {print("");}

////
//Test four variants of if-statement, WITH curly-braces
////
on akey {
  if(42 == 42)
  { print("Test-E, they are equal."); }
}

on akey {
  if(42 == 41)
  { print("Test-F, should not see this message."); }
}

on akey {
  if(42 == 41)
  { print("Test-G, should not see this message."); }
  else
  { print("Test-G, they are not equal."); }
}

on akey {
  if(42 == 42)
  { print("Test-H, they are equal."); }
  else
  { print("Test-H, should not see this message."); }
}

on akey {print("");}

////
//Test four variants of if-statement, WITH multiple statements IN curly-braces
////
on akey {
  if(42 == 42)
  {
    print("Test-I, they are equal.");
    print("Test-I, they are still equal.");
  }
}

on akey {
  if(42 == 41)
  {
    print("Test-J, should not see this message.");
    print("Test-J, still should not see this message.");
  }
}

on akey {
  if(42 == 41)
  {
    print("Test-K, should not see this message.");
    print("Test-K, still should not see this message.");
  }
  else
  {
    print("Test-K, they are not equal.");
    print("Test-K, they are still not equal.");
  }
}

on akey {
  if(42 == 42)
  {
    print("Test-L, they are equal.");
    print("Test-L, they are still equal.");
  }
  else
  {
    print("Test-L, should not see this message.");
    print("Test-L, still should not see this message.");
  }
}
