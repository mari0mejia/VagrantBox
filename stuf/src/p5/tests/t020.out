gpl.cpp::main()
  input file(tests/t020.gpl)
  random seed(42)
  read_keypresses_from_standard_input(true)
  dump_pixels(false)
  symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
gpl.cpp::main() Calling window->initialize().


gpl.cpp::main() Passing control to window->main_loop().
print[6]: Test-A, they are equal.
print[18]: Test-C, they are not equal.
print[23]: Test-D, they are equal.
print[28]: 
print[35]: Test-E, they are equal.
print[47]: Test-G, they are not equal.
print[52]: Test-H, they are equal.
print[57]: 
print[65]: Test-I, they are equal.
print[66]: Test-I, they are still equal.
print[86]: Test-K, they are not equal.
print[87]: Test-K, they are still not equal.
print[94]: Test-L, they are equal.
print[95]: Test-L, they are still equal.

User quit program. Printing the symbol table.
